packages/nimble-components/src/table/specs/table-column-width-hld.md

```diff
@@ -0,0 +1,170 @@
+- [#846 Interactively resize column width](https://github.com/ni/nimble/issues/846)
+- [Table README](./README.md)
+- [Table Design Doc](https://xd.adobe.com/view/5b476816-dad1-4671-b20a-efe796631c72-0e14/screen/d389dc1e-da4f-4a63-957b-f8b3cc9591b4/specs/)
+
+
+## Implementation / Design
+
+### Expected Behavior
+
+The behavior that has been prescribed for column sizing is as follows:
+
+- Columns should be able to be configured to either maintain a fixed width, or grow proportionally with the table such as when the window resizes causing the table width to increase. Tables can consist of columns that are configured as a mixture of the two modes.
+- If a user drags a divider between two columns to the right, then the column on the left will grow larger, and the column on the right will grow smaller by the same pixel amount. Sub-behaviors to this are:
+    - If a shrinking column has reached its minimum pixel size, then the next column in the direction of the sizing action will be affected up to the final column in a given direction.
+    - A sizing action to the left will ultimately stop having an effect when the left-most column reaches its minimum size.
+    - A sizing action to the right that would ultimately result in the final right column reaching its minimum size (all columns still within current table width) would then begin to push columns out of the table width resulting in a horizontal scrollbar.
+- Columns can be configured to not allow a user to interactively size them
+    - The implicit behavior present based on the behaviors described above, is that in a sizing action that cascades to a column configured to not be resized is that the column won't be resized towards a minimum size, and the cascade will effectively "skip" this column.
+
+### Out of Scope
+
+There are some column sizing behaviors that we will ultimately expect to support, but the APIs presented here are not meant to address:
+- Auto-resizing: We will not describe how we intend to support the use-case of having a column auto-size to its contents
+- Different interactive sizing modes: While the APIs described in this HLD do not inherently prescribe to a particular interactive sizing behavior, it's worth saying that in order to support multiple sizing modes, there will likely be additional APIs required that this HLD does not address.
+- Mechanisms related to accessibility-centric interactive column sizing (if there are such mechanisms). One possible example is allowing a user to size a column by way of the keyboard, instead of using a mouse. Ultimately, such a scenario is not in conflict with the API presented here, nor the mouse-based approach we know we will require, and can thus be handled separately, if ever.
+
+### API
+
+To accomodate the various sizing modes of a column, in addition to the other necessary behaviors, we can add the following properties to `TableColumn`:
+
+```
+class TableColumn {
```

Comment:
I think the reason is to distinguish between fractionalWidth (that the client specifies as an attribute) and the current calculated fractionalWidth that is the result of the fractionalWidth after a column size change.

So we wouldn't change the name of the fractionalWidth set by mixins so this name represents the current fractionalWidth as a part of user interaction. Some other options I can think of are `currentFractionalWidth`, `calculatedFractionalWidth`, `computedFractionalWidth`, `resolvedFractionalWidth`, `usedFractionalWidth` (just throwing all the CSS size adjectives in lol).

From CSS-isms I think this aligns most with "computed" values, i.e. it's the value after handling special cases (user interactions) but comes out as a relative unit, it is not the "used" value which is the exact pixel value used by the layout engine. So if going off css-isms I think I'd go for `computedFractionalWidth`.

But `computedFractionalWidth` may be confusing from a plugin API author perspective. Their plugin takes properties and assigns it to computedFractionalWidth / currentFractionalWidth. The table updates computedFractionalWidth / currentFractionalWidth on the plugin after the user column resize.

Reply:
So, my vote is to keep it as is. Personally, I think the meaning is accurate. While there is some calculation done to set `currentFractionalWidth/currentPixelWidth`, the value still represents what is _currently_ being used for layout, and, as @rajsite points out, I think it's more clear for a plugin (ideally mixin) author to be setting a property called `current<>` to the appropriate value, instead of `computed<>` or `calculated<>`.